interfaces/polkit/backend.go

```diff
@@ -70,8 +70,11 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	glob := polkitPolicyName(snapName, "*")
 	content := deriveContent(spec.(*Specification), snapInfo)
 	dir := dirs.SnapPolkitPolicyDir
-	if err := os.MkdirAll(dir, 0755); err != nil {
-		return fmt.Errorf("cannot create directory for polkit policy files %q: %s", dir, err)
+	if content != nil {
+		// If we do not have any content to write, there is no point ensuring the directory exists.
```

Comment:
nitpick: I think this comment makes more sense in the line above

Reply:
Fixed.